mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py

```diff
@@ -162,127 +57,129 @@ class mgmtsystem_nonconformity(base_state, orm.Model):
-        ),
-        'procedure_ids': fields.many2many(
-            'document.page', 'mgmtsystem_nonconformity_procedure_rel',
-            'nonconformity_id', 'procedure_id', 'Procedure'
-        ),
-        'description': fields.text('Description', required=True),
-        'state': fields.selection(_STATES, 'State', readonly=True),
-        'state_name': fields.function(
-            _state_name,
-            string='State Description',
-            type='char',
-            size=40,
-        ),
-        'system_id': fields.many2one('mgmtsystem.system', 'System'),
-        # 2. Root Cause Analysis
-        'cause_ids': fields.many2many(
-            'mgmtsystem.nonconformity.cause',
-            'mgmtsystem_nonconformity_cause_rel',
-            'nonconformity_id',
-            'cause_id',
-            'Cause',
-        ),
-        'severity_id': fields.many2one(
-            'mgmtsystem.nonconformity.severity',
-            'Severity',
-        ),
-        'analysis': fields.text('Analysis'),
-        'immediate_action_id': fields.many2one(
-            'mgmtsystem.action',
-            'Immediate action',
-            domain="[('nonconformity_id', '=', id)]",
-        ),
-        'analysis_date': fields.datetime('Analysis Date', readonly=True),
-        'analysis_user_id': fields.many2one(
-            'res.users',
-            'Analysis by',
-            readonly=True,
-        ),
-        # 3. Action Plan
-        'action_ids': fields.many2many(
-            'mgmtsystem.action',
-            'mgmtsystem_nonconformity_action_rel',
-            'nonconformity_id',
-            'action_id',
-            'Actions',
-        ),
-        'actions_date': fields.datetime('Action Plan Date', readonly=True),
-        'actions_user_id': fields.many2one(
-            'res.users',
-            'Action Plan by',
-            readonly=True,
-        ),
-        'action_comments': fields.text(
-            'Action Plan Comments',
-            help="Comments on the action plan.",
-        ),
-        # 4. Effectiveness Evaluation
-        'evaluation_date': fields.datetime('Evaluation Date', readonly=True),
-        'evaluation_user_id': fields.many2one(
-            'res.users',
-            'Evaluation by',
-            readonly=True,
-        ),
-        'evaluation_comments': fields.text(
-            'Evaluation Comments',
-            help="Conclusions from the last effectiveness evaluation.",
-        ),
-        # Multi-company
-        'company_id': fields.many2one('res.company', 'Company'),
-    }
-
-    _defaults = {
-        'company_id': (
-            lambda self, cr, uid, c:
-            self.pool.get('res.users').browse(cr, uid, uid, c).company_id.id),
-        'date': lambda *a: time.strftime(DATE_FORMAT),
-        'state': 'draft',
-        'author_user_id': lambda cr, uid, id, c={}: id,
-        'ref': 'NEW',
-    }
-
-    def create(self, cr, uid, vals, context=None):
+    # 1. Description
+    id = fields.Integer('ID', readonly=True)
```

Comment:
Yes, as far as I can say, the id should be created by default even on openerp 7 as it is used in ir_model_data. Not sure how it worked before but it shouldn't be necessary. I'll test it thought.